### ISSUE TITLE
resource/aws_glue_catalog_table: Prevent EntityNotFoundException when deleted outside of terraform

### ISSUE THREAD

**Issue by thomasv314** [terraform-providers/terraform-provider-aws#8173]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v0.11.13
+ provider.aws v2.4.0
```

### Affected Resource(s)

* aws_glue_catalog_table

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "us-west-2"
}

resource "aws_glue_catalog_database" "aws_glue_catalog_database" {
  name = "tvendettamycatalogdatabase"
}

resource "aws_glue_catalog_table" "aws_glue_catalog_table" {
  name          = "mycatalogtable"
  database_name = "tvendettamycatalogdatabase"
}
```

### Output

https://gist.github.com/thomasv314/9f2f61f1b741511f5101eaef7153f0f6

### Expected Behavior

Terraform should recreate a manually deleted aws_glue_catalog table

### Actual Behavior

Terraform throws an EntityNotFound exception

### Steps to Reproduce

1. `terraform apply`
2. Delete the AWS Glue Catalog Table manually in the console
3. `terraform plan`

### References

* #5136

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.6.0 of the Terraform AWS Provider, likely middle of next week. 👍

**Comment by bflad** [comment]
This has been released in [version 2.6.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#260-april-10-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
